### PR TITLE
Ports a couple of minor logging/admin related PRs from TG

### PR DESCRIPTION
**Pull request by Tyranicranger4**
## About The Pull Request

Ports/partially ports multiple minor logging PRs from TG, in addition to another minor admin related PR. Some of these are taken one to one - others are modified slightly; either for better readability (in my opinion) or to account for code differences.

Relevant PRs: 
https://github.com/tgstation/tgstation/pull/70096 Logging
https://github.com/tgstation/tgstation/pull/69206 Logging
https://github.com/tgstation/tgstation/pull/68819 Logging
https://github.com/tgstation/tgstation/pull/71399 Logging
https://github.com/tgstation/tgstation/pull/64867 Logging and also also swaps out the input list when aiming the BSA to a tgui input list (this change didn't originate from this PR, but that's where I'm taking it from)
https://github.com/tgstation/tgstation/pull/71681 This one isn't logging related, but was minor enough that I still threw it in. All it does is rename "End Round Normally" to "Undelay Round End" in the check-antagonists panel.

## Why It's Good For The Game
Logs are good. 
No reason not to have the BSA artillery input list be a tgui input list.
"Undelay Round End" is a much better way of wording the Undelay button.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

</details>

## Changelog
:cl: Tyranicranger4, San7890, Dragomagol, Melbert, Mothblocks
admin: Adds logging to Lavaland elite creation
admin: Adds logging to BSA aiming (seperate from firing)
admin: Adds logging to going DNR
admin: Adds logging to creating areas using blueprints (seperate from modifying them)
admin: Adds logging plus an admin notification for when non-roundstart upload consoles are created
admin: Renames the undelay round end button from "End Round Normally" to "Undelay Round End"
tweak: BSArtillery aiming now uses a tgui input list
/:cl:

### CODE REVIEW

File: code/__HELPERS/areas.dm [hunk 1]
```diff
@@ -107,6 +107,7 @@ GLOBAL_LIST_INIT(typecache_powerfailure_safe_areas, typecacheof(/area/engine/eng
 		FD.CalculateAffectingAreas()
 
 	to_chat(creator, "<span class='notice'>You have created a new area, named [newA.name]. It is now weather proof, and constructing an APC will allow it to be powered.</span>")
+	log_game("[key_name(creator)] created a new area: [AREACOORD(creator)] (previously \"[oldA.name]\")", LOG_GAME)
```
Comment (itsmeow) [inline code/__HELPERS/areas.dm]:
```suggestion
	log_game("[key_name(creator)] created a new area: [AREACOORD(creator)] (previously \"[oldA.name]\")")
```

File: code/game/machinery/computer/law.dm [hunk 2]
```diff
@@ -7,6 +7,9 @@
 /obj/machinery/computer/upload/Initialize(mapload)
 	. = ..()
 	AddComponent(/datum/component/gps, "Encrypted Upload")
+	if(!mapload)
+		log_game("A [name] was created at [loc_name(src)].")
```
Comment (itsmeow) [inline code/game/machinery/computer/law.dm]:
```suggestion
		log_game("A [name] was created at [AREACOORD(src)].")
```

File: code/modules/station_goals/bsa.dm [hunk 3]
```diff
@@ -304,8 +304,14 @@ DEFINE_BUFFER_HANDLER(/obj/machinery/bsa/middle)
...
-	var/V = input(user,"Select target", "Select target",null) in options|null
-	target = options[V]
+	var/victim = tgui_input_list(user, "Select target", "Artillery Targeting", options)
+	if(isnull(victim))
+		return
+	if(isnull(options[victim]))
+		return
+	target = options[victim]
+	var/datum/component/gps/log_target = target
+	log_game("[key_name(user)] has aimed the artillery strike at [get_area_name(log_target.parent)].")
```
Comment (itsmeow) [inline code/modules/station_goals/bsa.dm]:
```suggestion
	log_game("[key_name(user)] has aimed the bluespace artillery (BSA) strike at [AREACOORD(log_target.parent)].")
```

**Review by itsmeow — DISMISSED** [review]

File: code/modules/station_goals/bsa.dm (hunk 3 above)
Comment (Tyranicranger4) [reply code/modules/station_goals/bsa.dm]: get_area_name() is intentional here. At this point in time it's only gotten the area and has yet to select a tile, so I don't believe AREACOORD() will work here. I'll commit the readability improvement though

**Review by Rukofamicom — APPROVED** [review]
All looks good to me and requested changes were addressed a month ago now.

**Pull request merged**